app/knight/warior.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from app.knight_abilities.potion import Potion
+from app.knight_abilities.weapon import Weapon
+from app.knight_abilities.armor import Armour
+
+
+class Knight:
+    def __init__(self, name: str, dic_info: dict) -> None:
+        self.dic_info = dic_info
+        self.knight_info = self.dic_info[name]
```

Comment:
I could not claim that the following statement  is 100 %: it seems to me that the logic of looking up a specific knight (row 11) in the dictionary should be moved outside the class.

Reply:
no need to save all dict info in every instance of class Knight, u can just pass the info about several knight